src/lib/optimizer/strategy/column_pruning_rule.cpp

```diff
@@ -171,12 +171,12 @@ ExpressionUnorderedSet gather_locally_required_expressions(
     case LQPNodeType::Export:
     case LQPNodeType::Update:
     case LQPNodeType::ChangeMetaTable: {
-      const auto& left_input_expressions = node->left_input()->column_expressions();
-      locally_required_expressions.insert(left_input_expressions.begin(), left_input_expressions.end());
+      const auto& input_left_expressions = node->input_left()->column_expressions();
```

Comment:
Cases where we have `left_input_whatever` seem to be  difficult to automatically replace.